applications/FreeSurfaceApplication/custom_utilities/edgebased_gid_io.cpp

```diff
@@ -0,0 +1,5 @@
+#include <custom_utilities/edgebased_gid_io.h>
+
+namespace Kratos {
+    template class GidIO<EdgebasedGidGaussPointsContainer,EdgebasedGidMeshContainer>;
```

Comment:
I'd move the implementation of the functions in the header to the new cpp (maybe in a future PR to ease the review of this one).